### DISCUSSION TITLE
How to exit a nested sampling run that hangs

### DISCUSSION THREAD

**Discussion by amunozj** [Joshuaalbert/jaxns#155] [Q&A] [answered]
Hi friends!

I am using nested sampling to find the evidence associated with a model fit.   Under certain conditions, everything runs nicely, under others, it simply hangs after displaying the following message:

`Number of parallel Markov-chains set to: 50`

I have played around with all the input parameters to `DefaultNestedSampler`, as well as working in 64-bits, to no avail.  Is there a way to diagnose what the problem is? and or killing/stopping the call if it takes too long?

thanks a lot!

**Comment by amunozj** [comment, marked as answer]
My bad friends.  My calculation of `log_prob` inside my model was returning `-inf` in some cases and that was leading to the hang.  So if you run into this problem, check your log_prob :)

**Comment by Joshuaalbert** [comment]
Just in case others stumble on this there is a very helpful method for the model ```model.saneity_check(key, 100)```On Mar 20, 2024 00:17, Andres Muñoz-Jaramillo ***@***.***> wrote:
Closed #155 as resolved.
